File: translations/codenet/uncommented/C2Go/deepseek/Code/s176519935.go
package main

import (
	"fmt"
)

var heap []int64
var sz int

func push(x int) {
	i := sz
	sz++
	for i > 0 {
		p := (i - 1) / 2
		if heap[p] >= x {
			break
		}
		heap[i] = heap[p]
		i = p
	}
	heap[i] = x
}

func pop() int {
	ret := heap[0]
	x := heap[sz-1]
	sz--
	i := 0
	for i*2+1 < sz {
		a := i*2 + 1
		b := i*2 + 2
		if b < sz && heap[b] > heap[a] {
			a = b
		}
		if heap[a] <= x {
			break
		}
		heap[i] = heap[a]
		i = a
	}
	heap[i] = x
	return int(ret)
}

func main() {
	var n, m, i, j int
	var a []int64
	fmt.Scan(&n, &m)
	a = make([]int64, n)
	for i = 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	for i = 0; i < n; i++ {
		push(int(a[i]))
	}
	for i = 0; i < m; i++ {
		tm := int64(pop())
		tm /= 2
		push(int(tm))
	}
	var ans int64
	ans = 0
	for i = 0; i < n; i++ {
		ans += heap[i]
	}
	fmt.Println(ans)
}

File: translations/codenet/uncommented/C2P/codellama/Code/s176519935.py
#include<stdio.h>

heap = []
sz = 0

def push(x):
	i = sz
	while i > 0:
		p = (i - 1) / 2
		if heap[p] >= x:
			break
		heap[i] = heap[p]
		i = p
	heap[i] = x
	sz += 1

def pop():
	ret = heap[0]
	x = heap[sz - 1]
	i = 0
	while i * 2 + 1 < sz:
		a = i * 2 + 1
		b = i * 2 + 2
		if b < sz and heap[b] > heap[a]:
			a = b
		if heap[a] <= x:
			break
		heap[i] = heap[a]
		i = a
	heap[i] = x
	sz -= 1
	return ret

def main():
	n, m = map(int, input().split())
	a = list(map(int, input().split()))
	for i in range(n):
		push(a[i])
	for i in range(m):
		tm = pop()
		tm //= 2
		push(tm)
	ans = 0
	for i in range(n):
		ans += heap[i]
	print(ans)

if __name__ == "__main__":
	main()

